### DISCUSSION TITLE
Golang Function Util Folders

### DISCUSSION THREAD

**Discussion by steeltoken** [vercel/vercel#7000] [Help]
I get the error: Error! Could not find an exported function in "api/areas/common/consts.go"

when I try to include common code into their own files. Is it possible to somehow tag golang files/configure vercel deployment to ignore certain paths?

**Comment by ycjcl868** [comment]
same issues, for avoiding huge code all-in one go file.

**Comment by ainsleyclark** [comment]
Facing the same issue, did you find a fix?

**Comment by ainsleyclark** [comment]
If anyone comes across this, Vercel tries to parse all files in `api` that has a `http.HandlerFunc`. As a util or package `go` file doesn't usually contain this, you need to place an underscore under your pkg folder.

For example:
```
.
└── api
    ├── entrypoint.go
    └── _pkg
        └── util
            └── hello.go
```

For some reason it's not documented and I had to dig around in the Vercel tests.

For more info see here https://github.com/vercel/vercel/discussions/4983
